source/reference/string.rst

```diff
@@ -130,69 +130,162 @@ custom format types.  A JSON Schema validator will ignore any format
 type that it does not understand.
 
 .. index::
-   single: date-time
-   single: email
-   single: hostname
-   single: ipv4
-   single: ipv6
-   single: uri
-   single: uri-reference
-   single: uri-template
-   single: json-pointer
+   single: format
 
 Built-in formats
 ^^^^^^^^^^^^^^^^
 
 The following is the list of formats specified in the JSON Schema
 specification.
 
-- ``"date-time"``: Date representation, as defined by `RFC 3339, section
-  5.6 <http://tools.ietf.org/html/rfc3339#section-5.6>`_.
+.. index::
+   single: date-time
+   single: time
+   single: date
+   single: format; date-time
+   single: format; time
+   single: format; date
+
+Dates and times
+***************
+
+Dates and times are represented in `ISO8601 format
+<https://www.iso.org/iso-8601-date-and-time-format.html>`_.
+
+- ``"date-time"``: Date and time together, for example,
+  ``2018-11-13T20:20:39+00:00``.
+
+- ``"time"``: |draft7| Time, for example, ``20:20:39+00:00``
+
+- ``"date"``: |draft7| Date, for example, ``2018-11-13``.
+
+.. index::
+   single: email
+   single: idn-email
+   single: format; email
+   single: format; idn-email
+
+Email addresses
+***************
 
 - ``"email"``: Internet email address, see `RFC 5322,
   section 3.4.1 <http://tools.ietf.org/html/rfc5322#section-3.4.1>`_.
 
+- ``"idn-email"``: |draft7| The internationalized form of an Internet email address, see
+  `RFC 6531 <https://tools.ietf.org/html/rfc6531>`_.
+
+.. index::
+   single: hostname
+   single: idn-hostname
+   single: format; hostname
+   single: format; idn-hostname
+
+Hostnames
+*********
+
 - ``"hostname"``: Internet host name, see `RFC 1034, section 3.1
```

Comment:
Yes, punycode is an algorithm mapping full Unicode strings into the limited ASCII character set which is valid for hostnames in DNS.  Note that I said "international", not "internationalized".  I really should have just said "non-ascii" as that is not directly related to I18N or pertaining to multiple nations anyway.

This is a valid `"format": "idn-hostname"` which is not a valid `"format": "hostname"`:

`見.香港`

This is the same hostname, punycoded so that it is a valid `"format": "hostname"`, which, since all valid hostnames are valid idn-hostnames, is also a valid `"format": "idn-hostname"`:

`xn--nw2a.xn--j6w193g`

If you paste either into chrome's address bar (and presumably other browsers), it will assume HTTP and take you to the same web site.